tests/FunctionalTestCase.php

```diff
@@ -201,6 +205,21 @@ protected function assertSameObjectId($expectedObjectId, $actualObjectId): void
         $this->assertEquals((string) $expectedObjectId, (string) $actualObjectId);
     }
 
+    /**
+     * Asserts that two given documents match.
+     *
+     * Extra keys in the actual value's document(s) will be ignored.
+     *
+     * @param array|object $expectedDocument
+     * @param array|object $actualDocument
+     */
+    public static function assertDocumentsMatch($expectedDocument, $actualDocument, string $message = ''): void
```

Comment:
I explained in another thread why I'd rather not introduce legacy spec test classes into the main test suite, but even if this was to stay this would be a method to add on the base TestCase class since it doesn't depend on a server connection.

The base TestCase is also where the existing BSON assertions are.